scripts/main.js

```diff
@@ -0,0 +1,121 @@
+var plots;
+function sortOptions() {
+
+    var x_axis_str = "<option disabled selected value> - select x-axis - </option>";
+    for (var i = 0; i < sort_options.length; i++){
+        x_axis_str += "<option>" + sort_options[i] + "</option>";
+    }
+    $("#x-axis-search").empty().append(x_axis_str);
+
+    var y_axis_str = "<option disabled selected value> - select y-axis - </option>";
+    for (var i = 0; i < sort_options.length; i++){
+        y_axis_str += "<option>" + sort_options[i] + "</option>";
+    }
+    $("#y-axis-search").empty().append(y_axis_str);
+
+    $.getJSON(json_url, function(plot) {
+        plots = plot;
+        var filters_str = "";
+        var lookup = {};
+        var result = [];
+
+        for (var l = 0; l < filters.length; l++){
+            filters_str += "<label>Filter" + (l+2) + "</label>";
+            filters_str += "<select id='filter" + (l+2) + "'>";
+            filters_str += "<option disabled selected value> - " + filters[l] + " - </option>";
+            lookup = {};
+            result = [];
+            for (var item, i = 0; item = plots[i++];) {
+                console.log("hey");
+                var name = item[filters[l]];
+                if (!(name in lookup)) {
+                    lookup[name] = 1;
+                    result.push(name);
+                }
+            }
+            for (var k = 0; k < result.length; k++){
+                filters_str += "<option>" + result[k]+ "</option>";
+            }
+            filters_str += "</select>";
+        }
+        $("#sort-form").append(filters_str);
+    });
+}
+
+
+function sortPlots() {
+
+    var x_axis = $('#x-axis-search').find(":selected").text();
+    var y_axis = $('#y-axis-search').find(":selected").text();
+    var filter = $('#filter-type').find(":selected").text();
+    var filters_arr = [];
+    //for when there are more filters than normalization
+    for (var l = 0; l < filters.length; l++){
+        filters_arr.push($('#filter' + (l+2)).find(":selected").text());
+    }
+
+    var rows = [];
+    var columns = [];
+    var filtered = [];
+    var xy_sort = [];
+    var row_str;
+
+    var selected_options = "<h3> x-axis: " + x_axis + "</h3>";
+    selected_options += "<h3> y-axis: " + y_axis + "</h3>";
+    selected_options += "<h3> filter: " + filter + "</h3>";
+
+    for (var v = 0; v < filters.length; v++){
+        selected_options += "<h3> " + filters[v] + ": " + filters_arr[v] + "</h3>";
+    }
+    $(".selected-options").empty().append(selected_options);
+    // filter type being determined
+    // filter options are displayed through filter_type.js
+    sort_options.splice(sort_options.indexOf(x_axis), 1);
+    sort_options.splice(sort_options.indexOf(y_axis), 1);
+    var filter_type = sort_options[0];
+        if ((x_axis != y_axis)) {
```

Comment:
Indentation is wrong here and you should use `!==` instead of `!=`

Reply:
fixed